packages/rspack/src/plugins/generate-package-json-plugin.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  ExecutorContext,
+  detectPackageManager,
+  serializeJson,
+  type ProjectGraph,
+} from '@nx/devkit';
+import {
+  HelperDependency,
+  createLockFile,
+  createPackageJson,
+  getHelperDependenciesFromProjectGraph,
+  getLockFileName,
+  readTsConfig,
+} from '@nx/js';
+import { type Compiler, type RspackPluginInstance } from '@rspack/core';
+import { RawSource } from 'webpack-sources';
+
+const pluginName = 'GeneratePackageJsonPlugin';
+
+export class GeneratePackageJsonPlugin implements RspackPluginInstance {
```

Comment:
This is essentially identical to `@nx/webpack`'s `GeneratePackageJsonPlugin` but utilizing the `@rspack/core` dependencies.

https://github.com/nrwl/nx/blob/e29e9f90b6c5cf2a5b686327d68de15b873c7c4c/packages/webpack/src/plugins/generate-package-json-plugin.ts